mypy/dmypy_server.py

```diff
@@ -288,6 +295,8 @@ def cmd_run(self, version: str, args: Sequence[str]) -> Dict[str, object]:
                     return {'restart': 'plugins changed'}
         except InvalidSourceList as err:
             return {'out': '', 'err': str(err), 'status': 2}
+        except SystemExit as e:
+            return {'out': '', 'err': stderr.getvalue(), 'status': e.code}
```

Comment:
I just stumbled upon another edge case: `dmypy run -- --help` will exit without any output. This is because argparse intercepts `--help` (and `-h` etc.) and prints the full help text to stdout and exits with zero status. I'm not sure what to do about this -- I suppose you could catch stdout too? Or ignore the issue?

Reply:
It would be just another context manager to capture stdout (since I backported the contextlib stuff). I will play around with this. I should probably also not suppress the help header if we want people to be able to run help from `dmypy run`.